app/services/power_manager/bmc.rb

```diff
@@ -13,19 +13,58 @@ def ready?
 
     attr_reader :proxy
 
+    def has_poweraction_fix_3_17?
+      nic = host.bmc_nic
+
+      case nic&.subnet&.bmc&.has_capability?(:BMC, :poweraction_fix_3_17)
+      in nil
+        Foreman::Logging.logger('app').error(
+          "Failed to determine capability of BMC proxy. Cancel power operation against #{host.name} because associated BMC NIC, Subnet or BMC Proxy is not found."
+        )
+        raise ::Foreman::BMCFeatureException.new(
+          N_("BMC NIC associated to %{host}, or Subnet/BMC Proxy associated to the BMC NIC not found.") %
+          { host: host.name }
+        )
+      in true
+        true
+      in false
+        false
+      end
+    rescue ActiveRecord::ActiveRecordError => e
+      Foreman::Logging.exception(
+        "Failed to determine capability of BMC proxy. Cancel power operation against #{host.name} due to unexpected ActiveRecord behavior.",
+        e,
+        level: :error
+      )
+      raise
```

Comment:
Although a defensive approach is a good way to go, I don't think we need the `nil` check here.

Calling `host.power.action`:

* If the host doesn't have an associated `power`, the error will be catched `ERF42-9958 [Foreman::Exception]: Unknown power management support - can't continue`
* If the subnet doesn't have an BMC proxy, the error is `Foreman::BMCFeatureException: ERF88-9474 [Foreman::BMCFeatureException]: There is no proxy with BMC feature set up. Associate a BMC feature with a subnet.`

```suggestion
      smart_proxy = host.smart_proxies.with_features(:BMC).first
      
      return false unless smart_proxy      
      return smart_proxy.has_capability?(:BMC, :poweraction_fix_3_17)
```

What do you think?